Start the final project -- base requirements

jjs0sbw opened the pull request:
Just a start.

jjs0sbw committed:
Starting final project.

jjs0sbw committed:
Update README.md

jjs0sbw committed:
Update README.md

jjs0sbw committed:
Start to update repo

jjs0sbw committed:
Start to update repo files

jjs0sbw committed:
Update more repo files

jjs0sbw committed:
Update model folder and  files

jjs0sbw committed:
Updatel folders and  files

jjs0sbw commented:
My first cut at a complete program...

I would like to add an input section..

jjs0sbw commented:
You need to load mongodb by:
Load mongo database data using -- mongo idea_1API < ideaJSON.js

 npm install 

node index

open in browser at localhost:3000

jamdon commented on public/js/templates/ideaListItem.tpl:
```diff
@@ -0,0 +1,6 @@
+<a class="ideaLink" href="/ideas/<%= name %>">
```
you don't have a `name` property in your model attributes, so this ends up empty and clicking on a single idea doesn't route property

jjs0sbw replied:
I used :name as a command line option in the AppRouter.js file routes code..

However, I have changed it to ideaName ... and it still seems to work the same way..

I am probably missing something..

Thanks for your help..

jamdon commented:
It's looking good. I pointed out a small issue in one of your templates (comment inline)

jjs0sbw committed:
Fixed name error

jjs0sbw commented:
Cool.. thanks for your  help...

jamdon commented on public/js/controllers/AppController.js:
```diff
@@ -0,0 +1,64 @@
+  jqxhr.done(function() {
+    if (options.name) {
+      model = ideas.findWhere({
+        name: options.name
```
`name` popping up again when there's no name property in your model attributes

jamdon commented on public/js/routers/AppRouter.js:
```diff
@@ -0,0 +1,31 @@
+    this.appController.showAllRankings();
+  },
+  showIdeasByRanking: function (ranking) {
+    this.appController.showIdeasByRanking({ ranking: ranking });
```
Your appController doesn't have this method on it (check spelling).

jjs0sbw replied:
OK... fixed that one..

jamdon commented:
Left a couple more inline comments highlighting issues that are preventing your router from working correctly.

jjs0sbw committed:
Changed name to ideaName

jjs0sbw committed:
Changed ..Ranking to ..Rank

jjs0sbw commented:
James:

Thanks for the review...

I have addressed your comments...

It is not clear to me that anything changed...

Take care, be good to yourself and have fun,

Joe

On Tue, Jun 14, 2016 at 9:56 PM, jamdon notifications@github.com wrote:

> Left a couple more inline comments highlighting issues that are preventing
> your router from working correctly.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Final_Project/pull/5#issuecomment-226087913,
> or mute the thread
> https://github.com/notifications/unsubscribe/AAiG-EjbeSaKHxcv7F-eqGgBGqlP8qCxks5qL4YHgaJpZM4ISeqF
> .

## 

Joe Simpson
“Reasonable people adapt themselves to the world. Unreasonable people
attempt to adapt the world to themselves. All progress, therefore, depends
on unreasonable people.”
George Bernard Shaw

jamdon commented:
Well, it fixed your `ideas/:ideaName` route, but there's still an issue with the `idea_ranking/:ranking` route. You fixed the name of the appController method you call, but if you look at the method name you mapped for this route in your appRouter, you didn't change that one to match. Loading these routes from the server is how you tell if they've been fixed.

jjs0sbw committed:
Fixed yet another bug

jjs0sbw commented:
OK.. thanks for the review..

I need to find a software tool set to analyze these larger code groups..

Just miss too much reading by eye..

jjs0sbw commented:
James...

Thanks again for review..

I think this one is now fixed..

Take care, be good to yourself and have fun,

Joe

On Wed, Jun 15, 2016 at 8:47 PM, jamdon notifications@github.com wrote:

> Well, it fixed your ideas/:ideaName route, but there's still an issue
> with the idea_ranking/:ranking route. You fixed the name of your
> appController method, but if you look at the method name you mapped for
> this route, you didn't change that one to match.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Final_Project/pull/5#issuecomment-226381395,
> or mute the thread
> https://github.com/notifications/unsubscribe/AAiG-KbtzevXezR4K40g10rI1tPh957sks5qMMdVgaJpZM4ISeqF
> .

## 

Joe Simpson
“Reasonable people adapt themselves to the world. Unreasonable people
attempt to adapt the world to themselves. All progress, therefore, depends
on unreasonable people.”
George Bernard Shaw

jamdon commented:
Awesome. Everything works great! 👍

jjs0sbw commented:
James..

Thanks for your help...

Could not have done it with out your guidance..

Take care, be good to yourself and have fun,

Joe

On Wed, Jun 15, 2016 at 9:35 PM, jamdon notifications@github.com wrote:

> Awesome. Everything works great! 👍
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Final_Project/pull/5#issuecomment-226386154,
> or mute the thread
> https://github.com/notifications/unsubscribe/AAiG-HdfHWcBFZMOHEtMIdTrfknyX10rks5qMNKXgaJpZM4ISeqF
> .

## 

Joe Simpson
“Reasonable people adapt themselves to the world. Unreasonable people
attempt to adapt the world to themselves. All progress, therefore, depends
on unreasonable people.”
George Bernard Shaw
